SimCalorimetry/HGCalSimProducers/src/HGCDigitizer.cc

```diff
@@ -99,42 +100,172 @@ namespace {
+          unsigned_charge = (unsigned_charge_array[ihit] & HGCDigitizer::dataMask);
+          unsigned_time = (unsigned_time_array[ihit] & HGCDigitizer::dataMask);
+          p_time = logintpack::unpack16log(unsigned_time, minPackChargeLog, maxPackChargeLog, base);
+          p_charge = logintpack::unpack16log(unsigned_charge, minPackChargeLog, maxPackChargeLog, base);
+          const std::pair<float, float> charge_time_pair = std::make_pair(p_charge, p_time);
+          (simIt->second).hit_info[0][iSample] += p_charge;
+          if (iSample == 9) {
+            if (hitRefs_bx0[detId].empty()) {
+              hitRefs_bx0[detId].emplace_back(charge_time_pair);
+            } else {
+              if (p_time < hitRefs_bx0[detId].back().second) {
+                itr findPos = std::upper_bound(hitRefs_bx0[detId].begin(),
+                                               hitRefs_bx0[detId].end(),
+                                               std::make_pair(0.f, p_time),
+                                               [](const auto& i, const auto& j) { return i.second < j.second; });
+
+                itr insertedPos = findPos;
+                if (findPos == hitRefs_bx0[detId].begin()) {
+                  insertedPos = hitRefs_bx0[detId].insert(insertedPos, std::make_pair(p_charge, p_time));
+                } else {
+                  itr prevPos = findPos - 1;
+                  if (prevPos->second == p_time) {
+                    prevPos->first = prevPos->first + p_charge;
+                    insertedPos = prevPos;
+                  } else if (prevPos->second < p_time) {
+                    insertedPos =
+                        hitRefs_bx0[detId].insert(findPos, std::make_pair((prevPos)->first + p_charge, p_time));
+                  }
+                }
+
+                for (itr step = insertedPos; step != hitRefs_bx0[detId].end(); ++step) {
+                  if (step != insertedPos)
+                    step->first += p_charge;
+                }
+
+                hitOrder_monitor[detId] = true;
+
+              } else if (p_time == hitRefs_bx0[detId].back().second) {
+                hitRefs_bx0[detId].back().first += p_charge;
+              } else if (p_time > hitRefs_bx0[detId].back().second) {
+                hitRefs_bx0[detId].emplace_back(std::make_pair(hitRefs_bx0[detId].back().first + p_charge, p_time));
+              }
+            }
+          }
+        }
+      }
+    }
+
+    if (minbiasFlag) {
+      for (const auto& hitmapIterator : hitRefs_bx0) {
+        unsigned int detectorId = hitmapIterator.first;
+        auto simIt = simData.emplace(detectorId, HGCCellInfo()).first;
+        //auto h = hitOrder_monitor.emplace(detectorId).first;
+        const bool orderChanged = hitOrder_monitor[detectorId];
+        int waferThickness = getCellThickness(geom, detectorId);
+        float cell_threshold = tdcForToAOnset[waferThickness - 1];
+        std::vector<std::pair<float, float>> hitRec = hitmapIterator.second;
```

Comment:
```suggestion
        const std::vector<std::pair<float, float>>& hitRec = hitmapIterator.second;
```
does not seem to be modified